pkg/platform/kube/deployer.go

```diff
@@ -222,36 +224,79 @@ func (d *deployer) getFunctionPodLogs(namespace string, name string) string {
 			namespace,
 			name)
 	} else {
+		pod := functionPods.Items[0]
 
-		// iterate over pods and get their logs
-		for _, pod := range functionPods.Items {
-			podLogsMessage += "\n* " + pod.Name + "\n"
-
-			logsRequest, getLogsErr := d.consumer.kubeClientSet.CoreV1().Pods(namespace).GetLogs(pod.Name, &v1.PodLogOptions{}).Stream()
-			if getLogsErr != nil {
-				podLogsMessage += "Failed to read logs: " + getLogsErr.Error() + "\n"
-				continue
+		// get the latest pod
+		for _, currentPod := range functionPods.Items {
+			if pod.ObjectMeta.CreationTimestamp.Before(&currentPod.ObjectMeta.CreationTimestamp) {
+				pod = currentPod
 			}
+		}
+
+		// get the pod logs
+		podLogsMessage += "\n* " + pod.Name + "\n"
+
+		logsRequest, getLogsErr := d.consumer.kubeClientSet.CoreV1().Pods(namespace).GetLogs(pod.Name, &v1.PodLogOptions{}).Stream()
+		if getLogsErr != nil {
+			podLogsMessage += "Failed to read logs: " + getLogsErr.Error() + "\n"
+		}
 
-			scanner := bufio.NewScanner(logsRequest)
+		scanner := bufio.NewScanner(logsRequest)
 
-			// get only first MaxLogLines logs
-			for i := 0; i < MaxLogLines; i++ {
+		// get only first MaxLogLines logs
+		for i := 0; i < MaxLogLines; i++ {
 
-				// check if there's a next line from logsRequest
-				if scanner.Scan() {
+			// check if there's a next line from logsRequest
+			if scanner.Scan() {
+				currentLogLine, err := d.prettifyPodLog(scanner.Bytes())
+				if err != nil {
 
-					// read the current token and append to logs
-					podLogsMessage += scanner.Text()
-				} else {
-					break
+					// when it is unstructured just add the log as a text
+					podLogsMessage += scanner.Text() + "\n"
+					continue
 				}
-			}
 
-			// close the stream
-			logsRequest.Close() // nolint: errcheck
+				// when it is a processor log line
+				podLogsMessage += currentLogLine + "\n"
+			} else {
+				break
+			}
 		}
+
+		// close the stream
+		logsRequest.Close() // nolint: errcheck
 	}
 
-	return podLogsMessage
+	return common.FixEscapeChars(podLogsMessage)
+}
+
+func (d *deployer) prettifyPodLog(log []byte) (string, error) {
+	logStruct := struct {
+		Time    *string `json:"time"`
+		Level   *string `json:"level"`
+		Message *string `json:"message"`
+		More    *string `json:"more,omitempty"`
+	}{}
+
+	if err := json.Unmarshal(log, &logStruct); err != nil {
+		return "", err
```

Comment:
Try doing  `strconv.Unquote` before the `json.Unmarshal`. This will probably avoid the problem you hit and you can get rid of `FixEscapeChars` which is very specific to 2 quoted sequences
https://stackoverflow.com/questions/16846553/how-to-unmarshal-an-escaped-json-string-in-go

Reply:
Unquote doesn't work for the pod logs - it returns an error because it is not the syntax it expects

Reply:
Talked offline, understood there's problem with unstructured / partially unstructured log from unhandled exception - caught errors. Suggest to generalize a *bit* here (replace `\\` -> `\`) and fix the loggers in the runtime to consistently spew out json logs, so that `prettifyPodLogLine` will be able to handle them